src/VisualStudio/CSharp/Test/ProjectSystemShim/CPS/AdditionalPropertiesTests.cs

```diff
@@ -29,5 +32,49 @@ public void SetProperty_RootNamespace_CPS()
             static string DefaultNamespaceOfSingleProject(TestEnvironment environment)
                 => environment.Workspace.CurrentSolution.Projects.Single().DefaultNamespace;
         }
+
+        [WpfFact]
+        [Trait(Traits.Feature, Traits.Features.ProjectSystemShims)]
+        public void SetProperty_MaxSupportedLangVersion_CPS()
+        {
+            var catalog = TestEnvironment.s_exportCatalog.Value
+                .WithParts(
+                    typeof(CSharpParseOptionsChangingService));
+
+            var factory = ExportProviderCache.GetOrCreateExportProviderFactory(catalog);
+
+            using (var environment = new TestEnvironment(exportProviderFactory: factory))
+            using (var project = CSharpHelpers.CreateCSharpCPSProject(environment, "Test"))
+            {
+                Assert.True(CanApplyVersionChange(environment, LanguageVersion.CSharp8));
+
+                // Test max version is less than attempted version
```

Comment:
Can we turn this into a Theory and have as inputs the max supported, version, and a Boolean if CanApply returns true/false?

Reply:
(That also eliminates the local function.)